chain/polkadot/relay_chain_node.go

```diff
@@ -0,0 +1,327 @@
+	"github.com/libp2p/go-libp2p-core/peer"
+	"go.uber.org/zap"
+
+	"github.com/decred/dcrd/dcrec/secp256k1/v2"
+	"github.com/strangelove-ventures/ibctest/ibc"
+	"github.com/strangelove-ventures/ibctest/internal/dockerutil"
+)
+
+type RelayChainNode struct {
+	log      *zap.Logger
+	TestName string
+
+	Home  string
+	Index int
+
+	NetworkID    string
+	containerID  string
+	VolumeName   string
+	DockerClient *client.Client
+	Image        ibc.DockerImage
+
+	Chain             ibc.Chain
+	NodeKey           p2pCrypto.PrivKey
+	AccountKey        *schnorrkel.MiniSecretKey
+	StashKey          *schnorrkel.MiniSecretKey
+	Ed25519PrivateKey p2pCrypto.PrivKey
+	EcdsaPrivateKey   secp256k1.PrivateKey
+
+	api         *gsrpc.SubstrateAPI
+	hostWsPort  string
+	hostRpcPort string
+}
+
+type RelayChainNodes []*RelayChainNode
+
+const (
+	wsPort         = "27451/tcp"
+	rpcPort        = "27452/tcp"
+	prometheusPort = "27453/tcp"
+)
+
+var (
+	IndexedName = []string{"alice", "bob", "charlie", "dave", "ferdie"}
+
+	RtyAtt = retry.Attempts(10)
+	RtyDel = retry.Delay(time.Second * 2)
+	RtyErr = retry.LastErrorOnly(true)
+)
+
+var exposedPorts = map[nat.Port]struct{}{
+	nat.Port(wsPort):         {},
+	nat.Port(rpcPort):        {},
+	nat.Port(prometheusPort): {},
+}
+
+// Name returns the name of the test node
+func (p *RelayChainNode) Name() string {
+	return fmt.Sprintf("relaychain-%d-%s-%s", p.Index, p.Chain.Config().ChainID, dockerutil.SanitizeContainerName(p.TestName))
+}
+
+// HostName returns the docker hostname of the test container
+func (p *RelayChainNode) HostName() string {
+	return dockerutil.CondenseHostName(p.Name())
+}
+
+// Bind returns the home folder bind point for running the node
+func (p *RelayChainNode) Bind() []string {
+	return []string{fmt.Sprintf("%s:%s", p.VolumeName, p.NodeHome())}
+}
+
+// NodeHome returns the working directory within the docker image,
+// the path where the docker volume is mounted.
+func (p *RelayChainNode) NodeHome() string {
+	return fmt.Sprintf("/home/.%s", p.Chain.Config().Name)
+}
+
+// PeerID returns the public key of the node key for p2p.
+func (p *RelayChainNode) PeerID() (string, error) {
+	id, err := peer.IDFromPrivateKey(p.NodeKey)
+	if err != nil {
+		return "", err
+	}
+	return peer.Encode(id), nil
+}
+
+// GrandpaAddress returns the ss58 encoded grandpa (consensus) address
+func (p *RelayChainNode) GrandpaAddress() (string, error) {
+	pubKey, err := p.Ed25519PrivateKey.GetPublic().Raw()
+	if err != nil {
+		return "", fmt.Errorf("error fetching pubkey bytes: %w", err)
+	}
+	return EncodeAddressSS58(pubKey)
+}
+
+// AccountAddress returns the ss58 encoded account address
+func (p *RelayChainNode) AccountAddress() (string, error) {
+	pubKey := make([]byte, 32)
+	for i, mkByte := range p.AccountKey.Public().Encode() {
+		pubKey[i] = mkByte
+	}
+	return EncodeAddressSS58(pubKey)
+}
+
+// AccountAddress returns the ss58 encoded stash address
+func (p *RelayChainNode) StashAddress() (string, error) {
```

Comment:
nit: godoc and method name mismatch. `// AccountAddress` -> `// StashAddress`

Reply:
done